docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,124 @@
+then ``yarn workspaces run build``.
 
-The critical thing here is that each event is processed in isolation - the application does not need to keep any application-related state between each event. It is this that differentiates it from a state-driven application.
+.. TODO (drsk) this process is changing right now, make sure it is documented up to date here.
 
-To do this, the application should:
+Authentication
+~~~~~~~~~~~~~~
 
-1. Create a connection to the Transaction Service, and instantiate a stream handler to handle the new event stream. By default, this will read events from the beginning of the ledger. This is usually not what is wanted, as it may replay already processed transactions. In this case, the application can request the stream from the current ledger end. This will, however, cause any events between the last read point and the current ledger end to be missed. If the application must start reading from the point it last stopped, it must record that point and explicitly restart the event stream from there.
+When you deploy your application to a production ledger, you need to authenticate the identities of
+your users.
 
-2. Optionally, create a connection to the Command Submission Service to send any required commands back to the ledger.
+DAML ledgers support a unified interface for authentication of commands. Some DAML ledgers like for
+example https://projectdabl.com offer an integrated authentication service, but you can also use an
+external service provider for authentication like https://auth0.com. The DAML react libraries
+support interfacing with an authenticated DAML ledger. Simply initialize your ``DamlLedger`` object
+with the token obtained by an authentication service. How authentication works and the form of the
+required tokens is described in the :ref:`Authentication <authentication>` section.
 
-3. Act on the content of events (type, content) to perform any action required by the application e.g. writing a database record or generating and submitting a command.
+Developer workflow
+~~~~~~~~~~~~~~~~~~
 
-.. _state-driven-applications-1:
+The DAML SDK enables a local development environment with fast iteration cycles. If you run
+``daml-reload-on-change.sh`` of the ``create-daml-app``, a local DAML sandbox ledger is started that
+is updated with your most recent DAML code on any change. Next, you can start your frontend in
+development mode by changing to your ``ui`` directory and run ``yarn start``. This will reload your
+frontend whenever you make changes to it. You can add unit tests for your DAML models by writing
+:ref:`DAML scenarios <testing-using-scenarios>`. These will also be reevaluated on change.  A
+typical DAML developer workflow is to
 
-Structuring state-driven applications
-=====================================
+  #. Make a small change to your DAML data model
+  #. Optionally test your DAML code and with scenarios (add reference to scenario testing in the docs)
```

Comment:
```suggestion
  #. Optionally test your DAML code with scenarios 
```
Can you add a reference for scenario testing pointing to the appropriate docs page? That should be the very last thing 😄

Reply:
yes, that's a good idea.